hawkular-inventory-json-helper/src/test/java/org/hawkular/inventory/json/SerializationTest.java

```diff
@@ -571,6 +573,46 @@ public void testInventoryStructure() throws Exception {
         test(s);
     }
 
+    @Test
+    public void testIdentityHashTree() throws Exception {
+        InventoryStructure<?> s = InventoryStructure.Offline.of(Feed.Blueprint.builder().withId("feed").build())
+                .addChild(ResourceType.Blueprint.builder().withId("resourceType").build())
+                .addChild(MetricType.Blueprint.builder(GAUGE).withId("metricType").withUnit(MetricUnit.NONE)
+                        .withInterval(0L).build())
+                .addChild(Metric.Blueprint.builder().withId("metrics").withMetricTypePath("metricType").withInterval
+                        (0L).build())
+                .startChild(
+                        Resource.Blueprint.builder().withId("resource").withResourceTypePath("resourceType").build())
+                .addChild(Resource.Blueprint.builder().withId("childResource").withResourceTypePath("../.resourceType")
+                        .build())
+                .end()
+                .build();
+
+        IdentityHash.Tree t = IdentityHash.treeOf(s);
+
+        test(t);
+    }
+
+    @Test
+    public void testSyncHashTree() throws Exception {
+        InventoryStructure<?> s = InventoryStructure.Offline.of(Feed.Blueprint.builder().withId("feed").build())
+                .addChild(ResourceType.Blueprint.builder().withId("resourceType").build())
+                .addChild(MetricType.Blueprint.builder(GAUGE).withId("metricType").withUnit(MetricUnit.NONE)
+                        .withInterval(0L).build())
+                .addChild(Metric.Blueprint.builder().withId("metrics").withMetricTypePath("metricType").withInterval
+                        (0L).build())
+                .startChild(
+                        Resource.Blueprint.builder().withId("resource").withResourceTypePath("resourceType").build())
+                .addChild(Resource.Blueprint.builder().withId("childResource").withResourceTypePath("../.resourceType")
+                        .build())
+                .end()
+                .build();
```

Comment:
Not sure what you mean. If this is duplicated somewhere in the class then I can look to change it, but then again, these are tests, so the locality of test setup and execution plays a role here, too, IMHO.